pkg/importer/imageio-datasource.go

```diff
@@ -146,16 +178,7 @@ func (is *ImageioDataSource) Close() error {
 		err = is.readers.Close()
 	}
 	if is.imageTransfer != nil {
-		if itID, ok := is.imageTransfer.Id(); ok {
-			transfersService := is.connection.SystemService().ImageTransfersService()
-			_, err = transfersService.ImageTransferService(itID).Finalize().Send()
-			if err != nil {
-				_, cancelErr := transfersService.ImageTransferService(itID).Cancel().Send()
-				if cancelErr != nil {
-					klog.Errorf("Unable to finalize or cancel transfer! Disk may remain locked until inactivity timeout.\n%v\n%v", err, cancelErr)
-				}
-			}
-		}
+		cleanupTransfer(is.connection, is.imageTransfer)
```

Comment:
Should we check the err returned by cleanupTransfer here, like in the defer above?

Reply:
The defer before this does the cleanup immediately after the transfer is complete (or failed) instead of leaving it open through any further phases. It checks the error result and sets is.imageTransfer to nil specifically to skip this cleanupTransfer on line 181, so it doesn't make the same kind of sense to do the same thing here.